### PR TITLE
LPS-186527 Migrate select tags

### PR DESCRIPTION
**Pull request by Stephy6**
Motivation
=======
The motivation behind this ticket is to use a a standard item selector component for tags selector. 
[LPS-186527](https://liferay.atlassian.net/browse/LPS-186527)

Proposed Solution
========
What we did to solve this problem is to create an item selector implementation for tags and use it where is possible.

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by ealonso** [comment]
ci:test:sf

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by Stephy6** [comment]
ci:test:sf

**Comment by Stephy6** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
ci:test:sf

File: modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java [hunk 1]
```diff
@@ -1410,22 +1413,25 @@ public String getSocialBookmarksTypes() {
 
 	public String getTagSelectorURL() {
 		try {
```
Comment (ealonso) [inline modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java:1415]: @Stephy6 try/catch is not necessary

File: modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java [hunk 2]
```diff
@@ -1410,22 +1413,25 @@ public String getSocialBookmarksTypes() {
...
-			PortletURL portletURL = PortletProviderUtil.getPortletURL(
-				_httpServletRequest, AssetTag.class.getName(),
-				PortletProvider.Action.BROWSE);
+			RequestBackedPortletURLFactory requestBackedPortletURLFactory =
```
Comment (ealonso) [inline modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java:1416]: @Stephy6 you can inline it

File: modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java [hunk 3]
```diff
@@ -1410,22 +1413,25 @@ public String getSocialBookmarksTypes() {
...
+			AssetTagsItemSelectorCriterion assetTagsItemSelectorCriterion =
+				new AssetTagsItemSelectorCriterion();
 
-			portletURL.setParameter(
-				"groupIds", StringUtil.merge(getGroupIds()));
-			portletURL.setParameter(
-				"eventName", _portletResponse.getNamespace() + "selectTag");
-			portletURL.setParameter("selectedTagNames", "{selectedTagNames}");
-			portletURL.setWindowState(LiferayWindowState.POP_UP);
+			assetTagsItemSelectorCriterion.setDesiredItemSelectorReturnTypes(
```
Comment (ealonso) [inline modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java:1422]: @Stephy6 You should remove empty lines between assetTagsItemSelectorCriterion.setXXX and also it should be ordered alphabetically

File: modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java [hunk 4]
```diff
@@ -1410,22 +1413,25 @@ public String getSocialBookmarksTypes() {
...
+				new AssetTagsItemSelectorReturnType());
 
-			return portletURL.toString();
+			assetTagsItemSelectorCriterion.setMultiple(true);
```
Comment (ealonso) [inline modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java:1425]: @Stephy6 setMultiple method doesn't exist

**Comment by ealonso** [comment]
ci:test:sf

**Comment by ealonso** [comment]
see: https://github.com/liferay-echo/liferay-portal/pull/12815

**Pull request closed without merging**